core/src/test/scala/org/apache/spark/sql/delta/CheckpointsSuite.scala

```diff
@@ -68,6 +72,93 @@ class CheckpointsSuite extends QueryTest
       }
     }
   }
+
+  test("multipart checkpoints") {
+     withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+
+      withSQLConf(
+        DeltaSQLConf.DELTA_CHECKPOINT_MAX_ACTIONS_PER_FILE.key -> "10",
+        DeltaConfigs.CHECKPOINT_INTERVAL.defaultTablePropertyKey -> "1") {
+        // 3 actions, protocol, metaData, and new file
+        spark.range(1).repartition(1).write.format("delta").save(path)
+        val deltaLog = DeltaLog.forTable(spark, path)
+
+        // 4 total actions, 1 new file
+        spark.range(1).repartition(1).write.format("delta").mode("append").save(path)
+
+        assert(deltaLog.lastCheckpoint.isDefined)
+        deltaLog.lastCheckpoint.foreach { checkpointMetadata =>
+          assert(checkpointMetadata.version == 1)
+          assert(checkpointMetadata.parts.isEmpty)
+        }
+        val checkpointPath =
+          FileNames.checkpointFileSingular(deltaLog.logPath, deltaLog.snapshot.version).toUri
+        assert(new File(checkpointPath).exists())
+
+        // 11 total actions, 7 new files
+        spark.range(30).repartition(7).write.format("delta").mode("append").save(path)
+        deltaLog.lastCheckpoint.foreach { checkpointMetadata =>
+          assert(checkpointMetadata.version == 2)
+          assert(checkpointMetadata.parts.isDefined)
+          assert(checkpointMetadata.parts.get == 2)
+        }
```

Comment:
nit: these checks are repeated. May be an utility method will simplify the code?

Reply:
Yeah i'll try to make one

Reply:
Made helper for this